gc/src/trace.rs

```diff
@@ -17,6 +17,12 @@ pub trait Finalize {
     fn finalize(&self) {}
 }
 
+/// Ensures type has no custom Drop implementation, without implementing it itself
+/// In case of Drop implementation present, obscure compilation error will be raised
+pub unsafe trait TriviallyDrop {
+    unsafe fn guard(self);
```

Comment:
Although the trait should clearly be `unsafe`, the method needn’t be `unsafe`, since it’s just a fancy equivalent of `std::mem::drop`.

Do we need a trait for this though? We could just add an unused lambda function `let _trivially_drop = |t: Self| { match t { #trivially_drop_body } };` to the implementation of one of the `Trace` methods.

Reply:
It was unsafe because of older implementation of this check, which was unsafe dyn trait magic at runtime.

Well, i think trait is useless now, moved _TRIVIALLY_DROP to Finalize trait implementation.